LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/LoRaMessage/LoRaPayloadData.cs

```diff
@@ -393,13 +413,21 @@ public byte[] GetDecryptedPayload(string appSkey)
         /// <summary>
         ///  Replaces the <see cref="Frmpayload"/>, encrypting the values.
         /// </summary>
-        public override byte[] PerformEncryption(string appSkey)
+        public override byte[] PerformEncryption(NetworkSessionKey key) =>
+            PerformEncryption(GetDecryptedPayload(key));
+
+        /// <summary>
+        ///  Replaces the <see cref="Frmpayload"/>, encrypting the values.
+        /// </summary>
+        public override byte[] PerformEncryption(AppSessionKey key) =>
+            PerformEncryption(GetDecryptedPayload(key));
+
+        private byte[] PerformEncryption(byte[] rawDecryptedPayload)
```

Comment:
I don't get the name of this method. The method does copy the passed in byte buffer to the RawMessage. That's an operation that won't do encryption, as far as I understand.

Reply:
Ok - is there an overload that does? Otherwise I would suggest to change the name. You can resolve when you checked this.

Reply:
Renamed it to the same as its calling method in 7855f2d8eb479e2239f24ffc4c35bac85e380440. Please let me know if that's better.